### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Depaton**

### CODE REVIEW

**Comment by Depaton** [comment]
Screenshot_2024-01-26_03-44-54
Screenshot_2024-01-26_03-42-53
Screenshot_2024-01-26_03-46-03
Screenshot_2024-01-26_03-36-23

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+venv/
+.idea/
+pyc/
+db.sqlite3
```
Comment (vasylhnatiuk) [inline .gitignore]: add newline

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: taxi/models.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    country = models.CharField(max_length=240)
+
+    def __str__(self):
+        return self.name
+
+
+class Car(models.Model):
+    model = models.CharField(max_length=240)
+    manufacturer = models.ForeignKey(Manufacturer, on_delete=models.CASCADE)
+    drivers = models.ForeignKey(Driver, on_delete=models.SET_NULL, null=True)
```
Comment (pavlejviki) [inline taxi/models.py]: add related_name attribute to ForeignKey fields

**Review by pavlejviki — CHANGES_REQUESTED** [review]

File: taxi/models.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    manufacturer = models.ForeignKey(
+        Manufacturer,
+        on_delete=models.CASCADE,
+        related_name="Manufacturer"
+    )
+    drivers = models.ForeignKey(
+        Driver,
+        on_delete=models.SET_NULL,
+        null=True,
+        related_name="Drivers"
```
Comment (Dimosphen1) [inline taxi/models.py]: The `related_name` is picked up incorrectly for both fields, the related name should represent the way how model objects would be retrieved from the side of the related model, e.g. you would get all cars that relate to certain driver by referring to `cars` from the driver, not `Drivers`. Related names should be lowercase and preferably in a plural form. [Here](https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.ForeignKey.related_name) you can read more extensive documentation on this topic. Use `"cars"` as a `related_name` for both `drivers` and `manufacturer` fields.

File: taxi/models.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class Car(models.Model):
+    model = models.CharField(max_length=240)
+    manufacturer = models.ForeignKey(
+        Manufacturer,
+        on_delete=models.CASCADE,
+        related_name="Manufacturer"
+    )
+    drivers = models.ForeignKey(
```
Comment (Dimosphen1) [inline taxi/models.py]: `drivers` should have a many-to-many relationship, change the field to encounter that.

File: taxi/migrations/0003_rename_manufactured_car_manufacturer.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        ('taxi', '0002_manufacturer_car'),
+    ]
+
+    operations = [
+        migrations.RenameField(
+            model_name='car',
+            old_name='manufactured',
+            new_name='manufacturer',
+        ),
+    ]
```
Comment (Dimosphen1) [inline taxi/migrations/0003_rename_manufactured_car_manufacturer.py]: It is a good practice to include one migration per app per pull request in Django projects to not overload the `migrations` folder. Include one migration to this pull request, please.

File: tests/test_taxi.py [hunk 6]
```diff
@@ -69,7 +67,7 @@ def test_car_is_register_in_admin(self):
 
 class GitignoreTests(TestCase):
     def test_gitignore_exist(self):
-        file_exists = os.path.exists('.gitignore')
+        file_exists = os.path.exists(".gitignore")
         assert file_exists
 
     def test_gitignore_has_correct_content(self):
```
Comment (Dimosphen1) [inline tests/test_taxi.py:73]: Remove the changes made to the test file.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested.

**Pull request closed without merging**